packages/reactivity/src/ref.ts

```diff
@@ -235,7 +238,10 @@ export type ToRef<T> = IfAny<T, Ref<T>, [T] extends [Ref] ? T : Ref<T>>
 export function toRef<T extends object, K extends keyof T>(
   object: T,
   key: K
-): ToRef<T[K]>
+): {
+  0: ToRef<T[K]>
+  1: Readonly<ToRef<T[K]>>
+}[IfReadonlyKey<T, K>]
```

Comment:
Same here, and maybe we could extract it to a new type.

Reply:
Updated to use `true` / `false`. Not sure how to abstract this nicely. Suggestions?